### PR TITLE
fix: performance of base64 encode/decode large arrays

### PR DESCRIPTION
**Pull request by petersalomonsen**
contract calls with large Uint8Arrays fails with gas exceeded because base64 encoding/decoding was
too slow. this fix contains adjustments to increase performance of base64 encoding and decoding.

Without this fix I get the error:

```
{
  "ActionError": {
    "index": 0,
    "kind": {
      "FunctionCallError": {
        "HostError": "GasExceeded"
      }
    }
  }
}```

as can be found here.

https://explorer.testnet.near.org/transactions/7DRNWgsza9tRxU3QeYbkjpEoQLZouKfmJxirbk9QuBXL

### CODE REVIEW

**Comment by willemneal** [comment]
Great thanks!

**Comment by petersalomonsen** [comment]
Not sure what’s going on with the tests though. Seeing it’s failing in master too.

File: sdk-core/assembly/base64.ts [hunk 1]
```diff
@@ -2,115 +2,124 @@
...
-        break;
-    }
-
-    return x.join("");
-  }
-
-  function getByte64(s: string, i: u32): u32 {
-    return ALPHA.indexOf(s.charAt(i));
-  }
+const PADCHAR = "=";
```
Comment (willemneal) [inline sdk-core/assembly/base64.ts:6]:
```suggestion
// @ts-ignore: decorator
@lazy
const PADCHAR = "=";
```

File: sdk-core/assembly/base64.ts [hunk 2]
```diff
@@ -2,115 +2,124 @@
...
+const ALPHA = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
```
Comment (willemneal) [inline sdk-core/assembly/base64.ts:7]:
```suggestion
// @ts-ignore: decorator
@lazy
const ALPHA = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
```

File: sdk-core/assembly/base64.ts [hunk 3]
```diff
@@ -2,115 +2,124 @@
...
+ 
+const ALPHAVALUES = new Uint8Array(128);
```
Comment (willemneal) [inline sdk-core/assembly/base64.ts:9]:
```suggestion
// @ts-ignore: decorator
@lazy
const ALPHAVALUES = new Uint8Array(128);
```

**Review by willemneal — CHANGES_REQUESTED** [review]
Thanks for looking into this! There is a separate base64 library that we created, but never moved toward so perhaps this work could also help there.

Currently this fails a test that checks that compiling an empty file produces and empty binary.  This is because we want to make sure that things like unused globals don't end up in the binary, which is why we used the `@lazy` decorator.  Accepting my suggested changes should make them pass.

**Comment by petersalomonsen** [comment]
yeah ok. I've accepted your changes, makes sense, and still works with my contract :) 

should I rebase+squash too, or will you do it when merging?

**Review by willemneal — APPROVED** [review]

**Pull request merged**